packages/cozy-stack-client/src/CozyStackClient.js

```diff
@@ -107,6 +107,45 @@ class CozyStackClient {
     }
   }
 
+  /**
+   * Retrieves a new app token by refreshing the currently used token.
+   * @throws {Error} The client should already have an access token to use this function
+   * @throws {Error} The client couldn't fetch a new token
+   * @returns {Promise} A promise that resolves with a new AccessToken object
+   */
+  async refreshToken() {
+    if (!this.token) throw new Error('No token to refresh')
+
+    const options = {
+      method: 'GET',
+      credentials: 'include'
+    }
+    const response = await fetch('/', options)
+
+    if (!response.ok) {
+      throw new Error("couldn't fetch a new token - response ko")
+    }
+    const html = await response.text()
+    const parser = new DOMParser()
+    const doc = parser.parseFromString(html, 'text/html')
+    if (!doc) {
+      throw Error("couldn't fetch a new token - doc is not html")
+    }
+    const appNode = doc.querySelector('div[role="application"]')
+    if (!appNode) {
+      throw Error("couldn't fetch a new token - no div[role=application]")
+    }
+    const cozyToken = appNode.dataset.cozyToken
+    if (!cozyToken) {
+      throw Error("couldn't fetch a new token -- no cozyToken node")
```

Comment:
`s/no cozyToken node/missing data-cozy-token attribute/`, so one doesn't start wondering about some `#cozyToken` node for example?